dts/bindings/phy/st,stm32-otghs-phy.yaml

```
@@ -0,0 +1,27 @@
+# Copyright (c) 2024 Meta
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    This binding is to be used by the STM32U5xx transceivers which are built-in
+    with USB HS PHY IP and a configurable HSE clock source.
+
+compatible: "st,stm32-otghs-phy"
+
+include: phy-controller.yaml
+
+properties:
+  "#phy-cells":
+    const: 0
+
+  "clock-cfg":
+    type: int
+    enum:
+      - 0 # STM32U5_OTGHS_PHY_CLKSEL_16MHZ
+      - 1 # STM32U5_OTGHS_PHY_CLKSEL_19P2MHZ
+      - 2 # STM32U5_OTGHS_PHY_CLKSEL_20MHZ
+      - 3 # STM32U5_OTGHS_PHY_CLKSEL_24MHZ
+      - 4 # STM32U5_OTGHS_PHY_CLKSEL_26MHZ
+      - 5 # STM32U5_OTGHS_PHY_CLKSEL_32MHZ
```

We could get rid of most of the crust by aligning these values to the underlying HAL define, as done here:
https://github.com/zephyrproject-rtos/zephyr/blob/fb6c4427b8343bebc6fd6f5737bea74dfa2e3368/dts/bindings/power/st%2Cstm32wb0-pwr.yaml#L43-L58

In the driver, this becomes a simple concatenation:
https://github.com/zephyrproject-rtos/zephyr/blob/fb6c4427b8343bebc6fd6f5737bea74dfa2e3368/soc/st/stm32/stm32wb0x/soc.c#L61-L62


Change 1 (+ description should be updated):
```suggestion
      - 1
      - 2
      - 3
      - 4
      - 5
      - 6
```